crates/sui-types/src/execution.rs

```diff
@@ -218,13 +250,17 @@ impl Value {
             Value::Object(_) => false,
             Value::Raw(RawValueType::Any, _) => true,
             Value::Raw(RawValueType::Loaded { abilities, .. }, _) => abilities.has_copy(),
+            Value::Receiving(_, _) => false,
         }
     }
 
     pub fn write_bcs_bytes(&self, buf: &mut Vec<u8>) {
         match self {
             Value::Object(obj_value) => obj_value.write_bcs_bytes(buf),
             Value::Raw(_, bytes) => buf.extend(bytes),
+            Value::Receiving(id, version) => {
+                buf.extend(&Receiving::new(*id, *version).to_bcs_bytes())
```

Comment:
Do you need the & here? Kinda surprised you do
```suggestion
                buf.extend(Receiving::new(*id, *version).to_bcs_bytes())
```